base/iterators.jl

```diff
@@ -473,7 +473,12 @@ zip_iteratoreltype() = HasEltype()
 zip_iteratoreltype(a) = a
 zip_iteratoreltype(a, tail...) = and_iteratoreltype(a, zip_iteratoreltype(tail...))
 
-last(z::Zip) = getindex.(z.is, minimum(Base.map(lastindex, z.is)))
+function last(z::Zip)
+    IteratorSize(z) == SizeUnknown() &&
+        throw(ArgumentError("Cannot get last element of zipped iterators of undefined lengths"))
+    return nth(z, length(z))
```

Comment:
Changing 
`getindex.(z.is, minimum(Base.map(lastindex, z.is)))`
to
`nth(z, length(z))`
is a clear win and an easy merge. Thanks!